src/Fields/field_tuples.jl

```diff
@@ -1,30 +1,67 @@
 """
-    VelocityFields(arch, grid; u=zeros(arch, grid), v=zeros(arch, grid), w=zeros(arch, grid))
+    VelocityFields(arch, grid; [u, v, w])
 
 Return a NamedTuple with fields `u`, `v`, `w` initialized on the architecture `arch`
-and `grid`. Data arrays may be passed via the `u`, `v`, and `w` keyword arguments.
+and `grid`. Fields may be passed via the optional keyword arguments `u`, `v`, and `w`.
 """
-function VelocityFields(arch, grid; u=zeros(arch, grid), v=zeros(arch, grid), w=zeros(arch, grid))
-    u = XFaceField(arch, grid, u)
-    v = YFaceField(arch, grid, v)
-    w = ZFaceField(arch, grid, w)
+VelocityFields(arch, grid;
+    u = XFaceField(arch, grid, UVelocityBoundaryConditions(grid), zeros(arch, grid)),
+    v = YFaceField(arch, grid, VVelocityBoundaryConditions(grid), zeros(arch, grid)),
+    w = ZFaceField(arch, grid, WVelocityBoundaryConditions(grid), zeros(arch, grid))
+) = (u=u, v=v, w=w)
+
+"""
+    VelocityFields(arch, grid, bcs::NamedTuple)
+
+Return a NamedTuple with fields `u`, `v`, `w` initialized on the architecture `arch`
+and `grid`. Boundary conditions `bcs` may be specified via a named tuple of
+`FieldBoundaryCondition`s.
+"""
+function VelocityFields(arch, grid, bcs::NamedTuple)
+    u_bcs = :u ∈ keys(bcs) ? bcs[:u] : UVelocityBoundaryConditions(grid)
+    v_bcs = :v ∈ keys(bcs) ? bcs[:v] : VVelocityBoundaryConditions(grid)
+    w_bcs = :w ∈ keys(bcs) ? bcs[:w] : WVelocityBoundaryConditions(grid)
+
+    u = XFaceField(arch, grid, u_bcs, zeros(arch, grid))
+    v = YFaceField(arch, grid, v_bcs, zeros(arch, grid))
+    w = ZFaceField(arch, grid, w_bcs, zeros(arch, grid))
+
     return (u=u, v=v, w=w)
 end
 
 """
     TracerFields(arch, grid, tracer_names; kwargs...)
 
 Return a NamedTuple with tracer fields specified by `tracer_names` initialized as
-`CellField`s on the architecture `arch` and `grid`. Optional `kwargs` can be specified
-to assign data arrays to each tracer field.
+`CellField`s on the architecture `arch` and `grid`. Fields may be passed via optional
+keyword arguments `kwargs` for each field.
 """
 function TracerFields(arch, grid, tracer_names; kwargs...)
```

Comment:
Even with the docstring I am a little confused how this function works. Maybe an example would help?

Reply:
I added an example to the docstring.